packages/d2mini/src/operators/reduce.ts

```diff
@@ -45,73 +44,52 @@ export class ReduceOperator<K, V1, V2> extends UnaryOperator<[K, V1], [K, V2]> {
       const currOut = this.#indexOut.get(key)
       const out = this.#f(curr)
 
-      // Create maps for current and previous outputs
-      const newOutputMap = new Map<
-        string,
-        { value: V2; multiplicity: number }
-      >()
-      const oldOutputMap = new Map<
-        string,
-        { value: V2; multiplicity: number }
-      >()
+      // Create maps for current and previous outputs using values directly as keys
+      const newOutputMap = new Map<V2, number>()
+      const oldOutputMap = new Map<V2, number>()
 
       // Process new output
       for (const [value, multiplicity] of out) {
-        const valueKey = hash(value)
-        if (newOutputMap.has(valueKey)) {
-          newOutputMap.get(valueKey)!.multiplicity += multiplicity
-        } else {
-          newOutputMap.set(valueKey, { value, multiplicity })
-        }
+        const existing = newOutputMap.get(value) ?? 0
```

Comment:
How come this works? This is using a built-in map so if objects are structurally equivalent but are different instances then they will be counted separately, no?